lib/components/user/monitored-trip/trip-basics-pane.js

```diff
@@ -87,9 +89,25 @@ class TripBasicsPane extends Component {
   }
 
   render () {
-    const { errors, isCreating, itineraryExistence, values: monitoredTrip } = this.props
+    const {
+      canceled,
+      dirty,
+      errors,
+      isCreating,
+      isSubmitting,
+      itineraryExistence,
+      values: monitoredTrip
+    } = this.props
     const { itinerary } = monitoredTrip
 
+    // Prevent user from leaving when form has been changed,
+    // but don't show it when they click submit or cancel.
+    const unsavedChanges = dirty && !isSubmitting && !canceled
+    // Message changes depending on if the new or existing trip is being edited
+    const unsavedChangesMessage = `You have haven't saved your ${
+      isCreating ? 'new ' : ''
+    }trip yet. If you leave, ${isCreating ? 'it' : 'changes'} will be lost.`
```

Comment:
I might just generify this a bit to something like:
```suggestion
const unsavedChangesMessage = 'You have unsaved changes to this trip. If you leave before saving, your changes will be lost'
```